CONTRIBUTING.md

```diff
@@ -42,19 +42,35 @@ our [Style Guide](docs/StyleGuide.md).
 
     * Run the [test suite](#testing) and make sure everything passes.
 
-    * When committing, be sure to write good commit messages. Stylistically,
+    * When committing, be sure to write good commit messages according [these 
+      seven rules](https://chris.beams.io/posts/git-commit/#seven-rules). Doing 
+      git commit prints a message if any of the rules is violated.
+      Stylistically,
       we use commit message titles in the imperative tense, e.g., `Add
       merge-append query optimization for time aggregate`.  In the case of
       non-trivial changes, include a longer description in the commit message
       body explaining and detailing the changes.  That is, a commit message
       should have a short title, followed by a empty line, and then
       followed by the longer description.
 
+    * When committing, add which GitHub issue is fixed or closed by the commit. 
+      For example, if the commit fixes bug 123, add a line at the end of the 
+      commit message with  `Fixes #123`, if the commit implements feature 
+      request 321, add a line at the end of the commit message `Closes #321`.
+      This will be recognized by GitHub. It will close the corresponding issue 
+      and place a hyperlink under the number.
+
 * Push your changes to an upstream branch:
 
     * Make sure that each commit in the pull request will represent a
       logical change to the code, will compile, and will pass tests.
 
+    * Make sure that the pull request message contains all important 
+      information from the commit messages including which issues are
+      fixed and closed. If a pull request contains one commit only, then
+      repeating the commit message is preferable, which is done automatically
```

Comment:
```suggestion
      repeating the commit message is preferred, which is done automatically
```

Reply:
Thanks for improvement suggestion. Is it due something explained in https://english.stackexchange.com/a/129002 ?